src/neps/plot/tensorboard_eval.py

```diff
@@ -0,0 +1,522 @@
+import math
+import os
+import random
+import warnings
+from typing import List, Optional, Union
+
+import torch
+from torch.utils.tensorboard import SummaryWriter
+from torch.utils.tensorboard.summary import hparams
+
+
+class SummaryWriter_(SummaryWriter):
+    """
+    This class inherits from the base SummaryWriter class and provides modifications to improve the logging.
+    It simplifies the logging structure and ensures consistent tag formatting for metrics.
+
+    Changes Made:
+    - Avoids creating unnecessary subfolders in the log directory.
+    - Ensures all logs are stored in the same 'tfevent' directory for better organization.
+    - Updates metric keys to have a consistent 'Summary/' prefix for clarity.
+    - Improves the display of 'Loss' or 'Accuracy' on the Summary file.
+
+    Methods:
+    - add_hparams: Overrides the base method to log hyperparameters and metrics with better formatting.
+    """
+
+    def add_hparams(self, hparam_dict, metric_dict, global_step):
+        if not isinstance(hparam_dict, dict) or not isinstance(metric_dict, dict):
+            raise TypeError("hparam_dict and metric_dict should be dictionary.")
+        updated_metric = {}
+        for key, value in metric_dict.items():
+            updated_key = "Summary" + "/" + key
+            updated_metric[updated_key] = value
+        exp, ssi, sei = hparams(hparam_dict, updated_metric)
+
+        self.file_writer.add_summary(exp)
+        self.file_writer.add_summary(ssi)
+        self.file_writer.add_summary(sei)
+        for k, v in updated_metric.items():
+            self.add_scalar(tag=k, scalar_value=v, global_step=global_step)
+
+
+class tblogger:
```

Comment:
After reviewing the rest of this, some of the parameters here could use some documentation as things like `logger.logger_init_bool` and `logger.logger_bool` aren't very clear from usage why they are needed. You might come up with a better name by documenting it ;)

It also seems like `logger.logger_bool` is 1-1 correlated with `disable_logging`, is `.logger_bool` needed?

You can add documentation for these kind of variables in this way. I think NePs uses `mkdocs-material` which means this will render nicely for the class if documentation every gets up there.

```python
class tblogger:
    x: T
    """This x does blah."""
    
    y: K
    """This y is for booping the booper.
    
    !!! warning "In multi-worker setup"
    
        Not all tblogger classes across workers will boop the booper
        and this variable may vary between workers.
    """
```